libtransmission/rpcimpl.cc

```diff
@@ -1789,6 +1789,11 @@ static char const* sessionSet(
         tr_sessionSetQueueStalledEnabled(session, boolVal);
     }
 
+    if (tr_variantDictFindStrView(args_in, TR_KEY_default_trackers, &sv))
```

Comment:
I'm a little concerned about the idea of passing a tracker list as a single string in settings, e.g. when a users enters this into a textfield and presses return to make a newline, how do we serialize that in settings.json? JSON doesn't handle newlines embedded in strings.

OT1H we could possibly store this in settings.json as `[ [ "tier1url1", "tier1url2" ], [ "tier2url1", tier2url2" ] ... ]`. OTOH I don't actually know if the code would work, since right now we largely use settings.json as a glorified ini file with single-depth key/value pairs. Refactoring the settings code would be out-of-scope for this PR.

Just brainstorming -- what if these were stored in a hardcoded `${session->config_dir}/default-trackers.txt`, similar to how blocklists are stored now? That would sidestep the issue of how to encode this in settings.

Reply:
> Just brainstorming -- what if these were stored in a hardcoded ${session->config_dir}/default-trackers.txt, similar to how blocklists are stored now?

I know I'm just an end user, but I love this idea. I remember struggling trying to figure out how to add it properly when I first cherry picked the this PR into my local build. I was using transmission-daemon and ended up loading transmission-gtk to add the URLs through the GUI settings and then looking at the settings file to adjust it on my daemon settings.

Having it as a text file would make updates super easy too.

Reply:
> how do we serialize that in settings.json? JSON doesn't handle newlines embedded in strings.

Look at "~/.config/transmission/settings.json" - newlines are serialised as "\n" and it works just fine.

I've been using this patch for almost five years. It works as advertised.